webhook/resources/setting/validator.go

```diff
@@ -44,20 +48,42 @@ func (v *settingValidator) Create(request *admission.Request, newObj runtime.Obj
 }
 
 func (v *settingValidator) Update(request *admission.Request, oldObj runtime.Object, newObj runtime.Object) error {
-	setting, ok := newObj.(*longhorn.Setting)
+	oldSetting, ok := oldObj.(*longhorn.Setting)
 	if !ok {
-		return werror.NewInvalidError(fmt.Sprintf("%v is not a *longhorn.Setting", newObj), "")
+		return werror.NewInvalidError(fmt.Sprintf("oldOjb %v is not a *longhorn.Setting", oldObj), "")
+	}
+	newSetting, ok := newObj.(*longhorn.Setting)
+	if !ok {
+		return werror.NewInvalidError(fmt.Sprintf("newObj %v is not a *longhorn.Setting", newObj), "")
+	}
+
+	if !cmp.Equal(oldSetting.ApplicableEngines, newSetting.ApplicableEngines) {
+		return werror.NewInvalidError("applicableEngines cannot be modified", "applicableEngines")
+	}
```

Comment:
Could this potentially break future upgrades where engine support changes?

Reply:
Good catch! We shouldn't validate it.